README.md

```diff
@@ -1,240 +1,229 @@
+
+ # verify this url valid and exist by running the curl --head command as below
+ curl --head https://mydevomsagentsa.blob.core.windows.net/agentshellbundle/docker-cimprov-10.0.0-0.universal.x86_64.sh
+ ```
+  > Note: x.x.x-x is the version of the docker provider which is determined from version info in docker.version file
+3. Update the azure storage blob location of docker-cimprov-x.x.x-x.universal.x86_64.sh in setup.sh
+ ```
+ # update the setup.sh in this with the url in above step in place of  https://github.com/microsoft/Docker-Provider/releases/download/..
+ cd ~/Docker-Provider/kubernetes/linux/
+ ```
+4. Build the Docker image via below command
+```
+   docker build -t  <repo>/<imagename>:<imagetag> .
+```
+5. Push the Docker image to docker repo
+```
+   docker push  <repo>/<imagename>:<imagetag>
+```
+
+## Windows Agent
+> Note: To build the Windows Agent Image, you will need Windows 10 Pro or higher machine with Docker for Windows
+### Install Pre-requisites
+1. Install .Net Core SDK 2.2 or higher from https://dotnet.microsoft.com/download if you dont have installed already
+2. Install go1.14.1 if you havent installed already
+  ```
+  cd  %userprofile%
+  mkdir go
+  cd  %userprofile%\go
+  curl -LO https://dl.google.com/go/go1.14.1.windows-amd64.msi
+  # install go. default will get installed %SYSTEMDRIVE%\go
+  msiexec /i %userprofile%\go\go1.14.1.windows-amd64.msi
+  ```
+3. Install build dependencies
+```
+cd  %userprofile%
+mkdir gcctemp && cd gcctemp
+## download gcc for windows
+curl -LO https://github.com/jmeubank/tdm-gcc/releases/download/v9.2.0-tdm64-1/tdm64-gcc-9.2.0.exe
+## install gcc on windows
+%userprofile%\gcctemp\tdm64-gcc-9.2.0.exe
+```
+4. Install Docker for windows https://docs.docker.com/docker-for-windows/install/
+
+### Build Certificate Generator Source code and Out OMS Go plugin code
+1. Build Certificate generator source code in .NET and Out OMS Plugin code in Go lang  by running these commands in CMD shell
+```
+cd %userprofile%\Docker-Provider\build\windows # based on your repo path
+powershell -executionpolicy bypass -File .\Makefile.ps1 # trigger build and publish
+```
+### Build Docker Image
+
+1.  Navigate to below directory to build the docker image
+```
+  cd %userprofile%\Docker-Provider\kubernetes\windows # based on your repo path
+  docker build -t  <repo>/<imagename>:win-<imagetag> .
+
+```
+2. Push the Docker image to docker repo. For testing, you will be pushing to Docker hub
+```
+  cd %userprofile%\Docker-Provider\kubernetes\windows # based on your repo path
+  docker push  <repo>/<imagename>:<imagetag>
+```
+
+# Update Kubernetes yamls
+
+Navigate to Kubernetes directory and update the yamls with latest docker image of Linux and Windows Agent and other relevant updates.
+
+#  Deployment and Validation
+Deploy the Kubernetes yamls on to your Kubernetes cluster with Linux and Windows nodes and make sure no regressions or whatever
```

Comment:
> Does some documentation exist for what sort of tests one needs to run to test for regressions?
> If not, it doesn't have to be added in this PR - I'm just curious if it does then we should add it here.

> since we dont have any tests other than one which Dilip added for health. I think, you can help on adding tests for example ones which you and rashmi discussed to validate windows agent.

Reply:
> > Does some documentation exist for what sort of tests one needs to run to test for regressions?
> > If not, it doesn't have to be added in this PR - I'm just curious if it does then we should add it here.
> 
> > since we dont have any tests other than one which Dilip added for health. I think, you can help on adding tests for example ones which you and rashmi discussed to validate windows agent.

> since you will be taking care of this so marking this resolved.